packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-dart-visitor.test.ts

```diff
@@ -796,4 +796,154 @@ describe('AppSync Dart Visitor', () => {
+      const schema = /* GraphQL */ `
+        type SqlPrimary @refersTo(name: "sql_primary") @model {
+          id: Int! @primaryKey
+          content: String
+          related: SqlRelated @hasOne(references: ["primaryId"])
+        }
+  
+        type SqlRelated @refersTo(name: "sql_related") @model {
+          id: Int! @primaryKey
+          content: String
+          primaryId: Int! @refersTo(name: "primary_id") @index(name: "primary_id")
+          primary: SqlPrimary @belongsTo(references: ["primaryId"])
+        }
+      `;
+  
+      ['SqlPrimary', 'SqlRelated'].forEach(modelName => {
+        const generatedCode = getVisitor({
+          schema,
+          selectedType: modelName,
+          isTimestampFieldsAdded: true,
+          respectPrimaryKeyAttributesOnConnectionField: true,
+          transformerVersion: 2,
+        }).generate();
+        expect(generatedCode).toMatchSnapshot();
+      });
+    });
+  
+    test('sets the association to the references field for hasOne and hasMany', () => {
+      const schema = /* GraphQL */ `
+        type Primary @model {
+          id: ID! @primaryKey
+          relatedMany: [RelatedMany] @hasMany(references: ["primaryId"])
+          relatedOne: RelatedOne @hasOne(references: ["primaryId"])
+        }
+        
+        type RelatedMany @model {
+          id: ID! @primaryKey
+          primaryId: ID!
+          primary: Primary @belongsTo(references: ["primaryId"])
+        }
+        
+        type RelatedOne @model {
+          id: ID! @primaryKey
+          primaryId: ID!
+          primary: Primary @belongsTo(references: ["primaryId"])
+        }
+      `;
+      ['Primary', 'RelatedOne', 'RelatedMany'].forEach(modelName => {
+        const generatedCode = getVisitor({
+          schema,
+          selectedType: modelName,
+          isTimestampFieldsAdded: true,
+          respectPrimaryKeyAttributesOnConnectionField: true,
+          transformerVersion: 2,
+        }).generate();
+        expect(generatedCode).toMatchSnapshot();
+      });
+    });
+  
+    test('double linked references', () => {
+      const schema = /* GraphQL */ `
+        type Foo @model {
+          id: ID!
+          bar1: Bar @hasOne(references: ["bar1Id"])
+          bar2: Bar @hasOne(references: ["bar2Id"])
+        }
+        
+        type Bar @model {
+          id: ID!
+          bar1Id: ID
+          bar2Id: ID
+          foo1: Foo @belongsTo(references: ["bar1Id"])
+          foo2: Foo @belongsTo(references: ["bar2Id"])
+        }
+      `;
+  
+      ['Foo', 'Bar'].forEach(modelName => {
+        const generatedCode = getVisitor({
+          schema,
+          selectedType: modelName,
+          isTimestampFieldsAdded: true,
+          respectPrimaryKeyAttributesOnConnectionField: true,
+          transformerVersion: 2,
+        }).generate();
+        expect(generatedCode).toMatchSnapshot();
+      });
+    });
+  
+    test('hasMany with sortKeyFields on primary key', () => {
```

Comment:
Here and for other platforms: we should probably add one for hasOne with sortKeyFields as well, just to be safe?